esmvaltool/diag_scripts/weighting/climwip/io_functions.py

```diff
@@ -140,3 +140,32 @@ def read_model_data(datasets: list) -> tuple:
     return read_input_data(datasets,
                            dim='model_ensemble',
                            identifier_fmt='{dataset}_{ensemble}_{exp}')
+
+
+def read_model_data_ancestor(cfg, variable_group) -> tuple:
+    """Load model data from ancestor folder."""
+    filepath = io.get_ancestor_file(cfg, 'MODELS_' + variable_group + '.nc')
+    ancestor_ds = xr.open_dataset(filepath)
+
+    anc_da = ancestor_ds['__xarray_dataarray_variable__'].load()
+    anc_da = anc_da.rename(variable_group)
+
+    return anc_da, filepath
+
+
+def read_observation_data(datasets: list) -> tuple:
+    """Load observation data from list of metadata."""
+    return read_input_data(datasets,
+                           dim='obs_ensemble',
+                           identifier_fmt='{dataset}')
+
+
+def read_observation_data_ancestor(cfg, variable_group) -> tuple:
+    """Load model data from ancestor folder."""
+    filepath = io.get_ancestor_file(cfg, 'OBS_' + variable_group + '.nc')
+    ancestor_ds = xr.open_dataset(filepath)
+
+    anc_da = ancestor_ds['__xarray_dataarray_variable__'].load()
```

Comment:
This seems fragile, is there no other way to retrieve this variable other than to rely on a hidden key?

```suggestion
    anc_da = ancestor_ds['__xarray_dataarray_variable__'].load()
```

Reply:
Hm, as it is this is just the variable name in the file. So if we want to solve this it needs to be saved differently. At the moment its a direct call of `xarray.Dataarray.to_netcdf()` here
https://github.com/ESMValGroup/ESMValTool/blob/42ff49a4672f25fd44f1115d0596b07bf0064970/esmvaltool/diag_scripts/weighting/calculate_difference_variable_group.py#L57
which is not ideal I guess. It creates this key as the variable has no name:
http://xarray.pydata.org/en/stable/generated/xarray.DataArray.to_netcdf.html?highlight=to_netcdf

Reply:
I will resolve this in a separate step.